apps/namadillo/src/atoms/masp/atoms.ts

```diff
@@ -1,18 +1,36 @@
+import { Asset } from "@chain-registry/types";
 import { Balance } from "@heliaxdev/namada-sdk/web";
 import { accountsAtom, defaultAccountAtom } from "atoms/accounts/atoms";
-import { chainTokensAtom, nativeTokenAddressAtom } from "atoms/chain";
+import { nativeTokenAddressAtom, tokenAddressesAtom } from "atoms/chain";
 import { shouldUpdateBalanceAtom } from "atoms/etc";
+import { availableAssetsAtom } from "atoms/integrations";
 import { queryDependentFn } from "atoms/utils";
 import BigNumber from "bignumber.js";
+import { atom } from "jotai";
 import { atomWithQuery } from "jotai-tanstack-query";
+import { namadaAsset } from "registry/namadaAsset";
+import { unknownAsset } from "registry/unknownAsset";
 import { getSdkInstance } from "utils/sdk";
+import { findExpoent, sumDollars } from "./functions";
+import { fetchCoinPrices } from "./services";
+
+export type TokenBalance = {
+  address: string;
+  amount: string;
```

Comment:
I know this type is coming from somewhere else, but is there a reason why amount is a `string` instead of a `BigNumber` here? Also, maybe we should use the denom from Asset

Reply:
The `address` and `amount` are the original values returned from the sdk when we execute `sdk.rpc.queryBalance`. It returns `Array<[address:string, amount:string]>`

The processed data are the other values: `denom`, `asset`, `balance` (the BigNumber you are looking for, but considering the `expoent`), and `dollar`. Dollar is optional as it could not be calculated if it's missing the value from the CoinGecko api (`nam` case), or in case it's missing the CoinGecko api key

Effectively we don't use the values of `address` and `amount` on the components, so I can remove it from the  `TokenBalance` if you feel it'll be cleaner

Reply:
Gotcha. I think this is fine for now :)